ConvertEventsToMDTest and MaskMDTest fail on win7-debug sometimes

A couple of nights ago they passed. Last night they failed. I remember similar failures of ConvertEventsToMD happening occasionally for pull requests on other platforms. Keep an eye on these tests.

`MDAlgorithmsTest_MaskMDTest` fails like this:

```
01:27:37 MaskMD-[Warning] Invalid value for Dimensions: Number of dimension ids/names must be n * 3. The following names were given: Axis0, Axis1, 
01:27:37 MaskMD-[Warning] Invalid value for Extents: Number of extents must be 4. 
01:27:37 .MaskMD-[Notice] MaskMD started
01:27:37 Exit code 0xc0000417
```

And `MDAlgorithmsTest_ConvertEventsToMDTest` failed like this:

```
01:27:26 Running 1 testConvertToMD-[Notice] ConvertToMD started
01:27:26 MDWSTransform-[Notice] Can not obtain transformation matrix from the input workspace: testEvWS as no oriented lattice has been defined. 
01:27:26 ConvertToMD-[Notice] ConvertToMD successful, Duration 0.03 seconds
01:27:26 
01:27:26 In ConvertEventsToMDTest::testEventWS:
01:27:26 C:/Jenkins/workspace/master_clean_win7_debug/Framework/MDAlgorithms/test/ConvertEventsToMDTest.h:61: Error: Expected (900 == ws->getNPoints()), found (900 != 899)
01:27:26 Failed 1 of 1 test
01:27:26 Success rate: 0%
```

--- a/Framework/MDAlgorithms/src/MaskMD.cpp
+++ b/Framework/MDAlgorithms/src/MaskMD.cpp
@@ -318,12 +318,13 @@ std::map<std::string, std::string> MaskMD::validateInputs() {
     validation_output["Extents"] = messageStream.str();
   }
   // Check extent value provided.
-  for (size_t i = 0; i < nDimensionIds; ++i) {
+  for (size_t i = 0; (i < nDimensionIds) && ((i * 2 + 1) < extents.size());
+       ++i) {
     double min = extents[i * 2];
     double max = extents[(i * 2) + 1];
     if (min > max) {
       messageStream << "Cannot have minimum extents " << min
-                    << " larger than maximum extents " << max;
+                    << " larger than maximum extents " << max << ". ";
       validation_output["Extents"] = messageStream.str();
     }
   }

